test_readWriteCrontab fails when crontab is an empty file

Test test_readWriteCrontab fails when crontab is an empty file.


    $ crontab -e

Save an empty crontab.

    $ crontab -l | wc -l
    0

    $ LC_ALL=en_US.UTF-8 make common test-v
    ...
    test_readWriteCrontab (test.test_tools.TestTools) ... FAIL
    ...
    Stdout:
    DEBUG: [common/tools.py:1237 readCrontab] Read 1 lines from users crontab
    DEBUG: [common/tools.py:1274 writeCrontab] Wrote 2 lines to users crontab
    DEBUG: [common/tools.py:1237 readCrontab] Read 1 lines from users crontab

Congrats, perfect issue report with excellent minimal reproducible example :-)

I could reproduce the test failure on HEAD and Ubuntu 20.04 with Python 3.8.10 via `LC_ALL=en_US.UTF-8; make test-v`:

```
FAIL: test_readWriteCrontab (test.test_tools.TestTools)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/XXX/dev/backintime/common/test/test_tools.py", line 556, in test_readWriteCrontab
    self.assertEqual(len(newCrontab), len(oldCrontab) + 1)
AssertionError: 1 != 2

Stdout:
DEBUG: [common/tools.py:1237 readCrontab] Read 1 lines from users crontab
DEBUG: [common/tools.py:1274 writeCrontab] Wrote 2 lines to users crontab
DEBUG: [common/tools.py:1237 readCrontab] Read 1 lines from users crontab
```

Would be great if a developer here raises the hand to fix this (I have no more time this week), since it is so easy to reproduce...

--- a/common/tools.py
+++ b/common/tools.py
@@ -1234,6 +1234,8 @@ def readCrontab():
             return []
         else:
             crontab = [x.strip() for x in out.strip('\n').split('\n')]
+            if crontab == ['']:  # Fixes issue #1181 (line count of empty crontab was 1 instead of 0)
+                crontab = []
             logger.debug('Read %s lines from users crontab'
                          %len(crontab))
             return crontab

